File: src/main/java/com/goudourasv/data/users/UserEntity.java
package com.goudourasv.data.users;

import com.goudourasv.data.courses.CourseEntity;

import javax.persistence.*;
import java.util.Set;
import java.util.UUID;

@Entity(name = "Users")
public class UserEntity {
    @Id
    @GeneratedValue
    private UUID id;
    @Column(name = "first_name")
    private String firstName;
    @Column(name = "last_name")
    private String lastName;
    @Column(name = "e_mail")
    private String email;
    @ManyToMany(mappedBy = "userEntities")
    private Set<CourseEntity> courseEntities;


    public UUID getId() {
        return id;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public String getEmail() {
        return email;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public void setEmail(String email) {
        this.email = email;
    }
}

File: src/main/java/com/goudourasv/data/users/UserEntity.kt
package com.goudourasv.data.users

import com.goudourasv.data.courses.CourseEntity
import com.goudourasv.data.ratings.RatingEntity
import java.util.*
import javax.persistence.*

@Entity(name = "Users")
class UserEntity(
    @Id
    @GeneratedValue
    var id: UUID? = null,

    @Column(name = "first_name")
    var firstName: String,

    @Column(name = "last_name")
    var lastName: String,

    @Column(name = "e_mail")
    var email: String,

    @ManyToMany(mappedBy = "favouritedByUsers")
    var courseEntities: MutableSet<CourseEntity>? = mutableSetOf(),

    @OneToMany(mappedBy = "userEntity", cascade = [CascadeType.ALL], orphanRemoval = true)
    var ratingEntities: MutableList<RatingEntity>? = mutableListOf(),

)
